Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/activities/EmulationActivity.java

```diff
@@ -915,6 +923,63 @@ private void chooseController()
     alertDialog.show();
   }
 
+  public static void handleWiiOverlayDisablesControllersSetting(int selection, boolean IsFirstRun)
+  {
+    if (NativeLibrary
+            .GetConfig(SettingsFile.FILE_NAME_DOLPHIN + ".ini", Settings.SECTION_INI_ANDROID,
+                    SettingsFile.KEY_WII_OVERLAY_DISABLES_CONTROLLERS, "False")
+            .equals("True"))
+    {
+      // TODO: Add real controllers to Wii Input overlay so they can be changed by this setting.
+      if (!NativeLibrary
+              .GetConfig(SettingsFile.FILE_NAME_DOLPHIN + ".ini", Settings.SECTION_INI_CORE,
+                      SettingsFile.KEY_GCPAD_PLAYER_1, InputOverlay.EMULATED_GAMECUBE_CONTROLLER)
+              .equals(InputOverlay.GAMECUBE_ADAPTER))
+      {
+        NativeLibrary
+                .SetConfig(SettingsFile.FILE_NAME_DOLPHIN + ".ini", Settings.SECTION_INI_CORE,
+                        SettingsFile.KEY_GCPAD_PLAYER_1,
+                        selection == InputOverlay.OVERLAY_GAMECUBE ?
+                                InputOverlay.EMULATED_GAMECUBE_CONTROLLER :
+                                InputOverlay.DISABLED_GAMECUBE_CONTROLLER);
+
+        NativeLibrary.SetOverlaySIDevice(selection == InputOverlay.OVERLAY_GAMECUBE ?
+                Integer.parseInt(InputOverlay.EMULATED_GAMECUBE_CONTROLLER) :
+                Integer.parseInt(InputOverlay.DISABLED_GAMECUBE_CONTROLLER));
+      }
+
+      if (!NativeLibrary
+              .GetConfig(SettingsFile.FILE_NAME_WIIMOTE + ".ini", SettingsFile.KEY_WIIMOTE_PLAYER_1,
+                      SettingsFile.KEY_WIIMOTE_TYPE, "1").equals("2"))
+      {
+        NativeLibrary.SetConfig(SettingsFile.FILE_NAME_WIIMOTE + ".ini",
+                SettingsFile.KEY_WIIMOTE_PLAYER_1, SettingsFile.KEY_WIIMOTE_TYPE,
+                selection == InputOverlay.OVERLAY_GAMECUBE ? "0" : "1");
+      }
+    }
+    // The following code is unnecessary on first run.
+    else if (!IsFirstRun)
+    {
+      if (selection == InputOverlay.OVERLAY_GAMECUBE)
+      {
+        NativeLibrary
+                .SetConfig(SettingsFile.FILE_NAME_DOLPHIN + ".ini",
+                        Settings.SECTION_INI_CORE,
+                        SettingsFile.KEY_GCPAD_PLAYER_1,
+                        InputOverlay.EMULATED_GAMECUBE_CONTROLLER);
+
+        NativeLibrary
+                .SetOverlaySIDevice(
+                        Integer.parseInt(InputOverlay.EMULATED_GAMECUBE_CONTROLLER));
```

Comment:
This needs error handling.

Reply:
Please check to make sure I updated this properly.